### PR TITLE
Graceful fossil error handling

### PR DESCRIPTION
**Pull request by Oliveriver**
**Brief Description of What This PR Does**

Ignores corrupt fossil files to prevent crashes when opening the Thriveopedia.

I went for the nuclear option of just wrapping the whole loading process in a try-catch block and printing the error. That may lead to error spam if a user with a corrupt fossil repeatedly opens and closes the Thriveopedia.

**Related Issues**

Closes https://github.com/Revolutionary-Games/Thrive/issues/3861

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/thriveopedia/fossilisation/FossilisedSpecies.cs [hunk 1]
```diff
@@ -119,15 +119,24 @@ public static bool IsSpeciesAlreadyFossilised(string name, List<string>? existin
     /// </summary>
     /// <param name="fossilName">The name of the .thrivefossil file (including extension)</param>
     /// <returns>The species saved in the provided file</returns>
```
Comment (Kasterisk) [inline src/thriveopedia/fossilisation/FossilisedSpecies.cs:121]: Can you update the `returns` documentation please?

**Review by Kasterisk — CHANGES_REQUESTED** [review]
I still like the approach of allowing corrupt entries just for the sake of telling players they have corrupt fossils but, this is good enough I guess.

File: src/thriveopedia/fossilisation/FossilisedSpecies.cs (hunk 1 above)
Comment (Oliveriver) [reply src/thriveopedia/fossilisation/FossilisedSpecies.cs]: Done.

**Comment by Oliveriver** [comment]
> I still like the approach of allowing corrupt entries just for the sake of telling players they have corrupt fossils but, this is good enough I guess.

It would certainly be possible to update that in future using a null check to display something else.

**Review by Kasterisk — APPROVED** [review]
That's good, as for now, this looks fine to me

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**